doc/api/stream.md

```diff
@@ -1608,6 +1621,41 @@ async function run() {
 run().catch(console.error);
 ```
 
+```js
+const pipeline = util.promisify(stream.pipeline);
+const fs = require('fs').promises;
+
+async function run() {
+  await pipeline(
+    async function*() {
+      const fd = await fs.open('archive.tar');
+      try {
+        const chunk = new Buffer(1024);
+        const { bytesRead } = await fs.read(fd, chunk, 0, chunk.length, null);
+        if (bytesRead === 0) return;
+        yield chunk.slice(0, bytesRead);
+      } finally {
+        await fs.close(fd);
+      }
+    },
+    zlib.createGzip(),
+    async function(source) {
+      const fd = await fs.open('archive.tar', 'w');
+      try {
+        for await (const chunk of source) {
+          await fs.write(fd, chunk);
+        }
+      } finally {
+        await fs.close(fd);
+      }
+    }
+  );
+  console.log('Pipeline succeeded.');
+}
+
+run().catch(console.error);
+```
```

Comment:
Can you add an example that goes along the lines of:

```js
pipeline(fs.createReadStream(), asyncGenTransform, fs.createWriteStream())
```

I think that would be the most common case.

Reply:
Updated example